thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DataProvider.java

```diff
@@ -136,4 +139,15 @@
    * @throws Exception
    */
   DetectionPipeline loadPipeline(DetectionConfigDTO config, long start, long end) throws Exception;
+
+  /**
+   * Returns a multimap of evaluations (keyed by the evaluations slice) for a given set of evaluations slices.
+   *
+   * @see Evaluation
+   *
+   * @param evaluationSlices the evaluation slices
+   * @param configId configId
+   * @return a multimap of evaluations (keyed by the evaluations slice)
+   */
+  Multimap<EvaluationSlice, EvaluationDTO> fetchEvaluations(Collection<EvaluationSlice> evaluationSlices, long configId);
```

Comment:
What if there are multiple rules in one configId? How do we handle this?

Reply:
Good point. This data fetcher will fetch the evaluations for all the rules. Maybe when evaluating models, we should group by the rule names as well.